tests/pruners_tests/test_hyperband.py

```diff
@@ -216,3 +216,27 @@ def objective(trial: optuna.trial.Trial) -> float:
 def test_incompatibility_between_bootstrap_count_and_auto_max_resource() -> None:
     with pytest.raises(ValueError):
         optuna.pruners.HyperbandPruner(max_resource="auto", bootstrap_count=1)
+
+
+def test_hyperband_pruner_and_grid_sampler() -> None:
+    pruner = optuna.pruners.HyperbandPruner(
+        min_resource=MIN_RESOURCE, max_resource=MAX_RESOURCE, reduction_factor=REDUCTION_FACTOR
+    )
+
+    search_space = {"x": [-50, 0, 50], "y": [-99, 0, 99]}
+    sampler = optuna.samplers.GridSampler(search_space)
+
+    study = optuna.study.create_study(sampler=sampler, pruner=pruner)
+
+    def objective(trial: optuna.trial.Trial) -> float:
+        for i in range(N_REPORTS):
+            trial.report(i, step=i)
+
+        x = trial.suggest_uniform("x", -100, 100)
+        y = trial.suggest_int("y", -100, 100)
+        return x ** 2 + y ** 2
+
+    study.optimize(objective)
```

Comment:
Not a strong opinion, but I'd like to set a large enough value to `n_trials`  in order to ensure that this test stops. If a developer happens to remove the `study.stop` from `GridSampler`, it will not stop by itself. I guess assertion errors will be more helpful for developers and CI.

```suggestion
    study.optimize(objective, n_trials=10)
```